### ISSUE TITLE
Investigate support for new protocols

### ISSUE THREAD

**Issue by psiinon** [zaproxy/zaproxy#4433]
Eg see https://www.theinquirer.net/inquirer/news/3025478/firefox-59-will-support-decentralised-internet-protocols

If anyone is interested in working on any of the specific protocols then we can raise specific bugs for each of them - the is more of a placeholder / reminder.

**Comment by palash25** [comment]
Is this a potential project for Code Sprint?

**Comment by psiinon** [comment]
Google Summer of Code or anything similar, yes :)

**Comment by palash25** [comment]
Thanks for the prompt reply. Will OWASP be organizing Code Sprint this summer?

**Comment by psiinon** [comment]
I suspect not, as its been accepted for GSoC.

**Comment by kingthorin** [comment]
Similar to websockets?

Would also probably require: [Enhancement: Add support for raising alerts for other than HTTP/1.x interactions](https://github.com/zaproxy/zaproxy/issues/2464)

**Comment by psiinon** [comment]
Yeah. The first part of any project would be working our the full set of changes we'd need for any specific protocol, and then we can prioritize the features e.g:

1. Proxy the protocol
1. Show the protocol in the GUI
1. Expose the protocol via the API
1. Support break
1. Support fuzzing
1. Support manually raising alerts
1. Support scanning

**Comment by juhakivekas** [comment]
There’s been a recent rise in things using WebRTC (mostly integrations in my experience). I don’t know if this is relevant in the usual testing scenarios, but at the moment not all apps using WebRTC work through zap, which means not even all HTTP traffic is testable with zap because some areas of the app are out of reach. This is a bit unfortunate, but either way simply proxying a protocol already has a lot of value.
